buildspec.yml

```diff
@@ -24,17 +24,26 @@ phases:
   pre_build:
     commands:
       - echo "Exporting of the image tag for building and pushing purposes"
+      # Reading git branch (the utility in the deploy script is unable to read it automatically on CodeBuild)
+      # see https://stackoverflow.com/questions/47657423/get-github-git-branch-for-aws-codebuild
+      - export GITHUB_BRANCH="$(git symbolic-ref HEAD --short 2>/dev/null)"
+      - |
+        if [ "${GITHUB_BRANCH}" = "" ] ; then
+          GITHUB_BRANCH="$(git branch -a --contains HEAD | sed -n 2p | awk '{ printf $1 }')";
+          export GITHUB_BRANCH=${GITHUB_BRANCH#remotes/origin/};
+        fi
       - COMMIT_HASH=$(echo $CODEBUILD_RESOLVED_SOURCE_VERSION | cut -c 1-7)
-      - GITHUB_BRANCH=${CODEBUILD_WEBHOOK_HEAD_REF#refs/heads/*}
-      # Manual build don't have the CODEBUILD_WEBHOOK_HEAD_REF variable set, therefore use the source version set during the manual build
-#      - GITHUB_BRANCH=${GITHUB_BRANCH:-${CODEBUILD_SOURCE_VERSION#pr/*}}
-      - export IMAGE_TAG="${GITHUB_BRANCH}.${COMMIT_HASH}"
+      - echo "GITHUB_BRANCH=${GITHUB_BRANCH} COMMIT_HASH=${COMMIT_HASH}"
       - make clean setup
 
   build:
     commands:
       - echo Build started on $(date)
-      - docker build -t ${IMAGE_BASE_NAME}:${IMAGE_TAG} .
+      - export DOCKER_IMG_TAG_BASE=${IMAGE_BASE_NAME}:${GITHUB_BRANCH}
+      - export DOCKER_IMG_TAG_BRANCH=${DOCKER_IMG_TAG_BASE}.${COMMIT_HASH}
```

Comment:
I would rename the variable DOCKER_IMG_TAG_BRANCH to DOCKER_IMG_TAG_HASH or DOCKER_IMG_TAG_COMMIT, this way the difference between DOCKER_IMG_TAG_HASH/COMMIT and DOCKER_IMG_TAG_LATEST is more explicit